components/form/TextField.tsx

```diff
@@ -80,7 +99,7 @@ export const TextField = React.forwardRef<RNView, TextFieldProps>(({name, label,
             onBlur={onBlur}
             onFocus={onFocus}
             onChangeText={onChangeText}
-            value={field.value as string} // TODO(skuznets): determine why the generics here don't collapse to string itself ...
+            value={field.value}
```

Comment:
It now knows the field.value is constrained to a `string` type. Or more specifically `string | null | undefined`.